### ISSUE TITLE
[DOCS] Flowcharts for intro docs

### ISSUE THREAD

**Issue by jsignell** [dask/dask#5965]
As discussed at the dask dev meeting it would be nice to have some visual representations of the top-level concepts presented in the dask docs. 

Specifically, the sci-kit learn flowchart was referenced as a helpful guide. 

Here is what we came up with at the meeting:
![image](https://user-images.githubusercontent.com/4806877/75697735-83314b00-5c7b-11ea-88cf-762bd67a9711.png)

And here is my first attempt at vectorizing it:
[![What API to Use](https://user-images.githubusercontent.com/4806877/75697681-6c8af400-5c7b-11ea-9b6d-6b2f68c3cd9b.png)](https://docs.google.com/drawings/d/1T3VJlW-IOGFNyvkflgZu-l8OCjxXWA4Lgo4dh4AWCs8/edit)

The other one that we thought might be useful is this one:
![image](https://user-images.githubusercontent.com/4806877/75697924-c55a8c80-5c7b-11ea-82a8-eec373c14532.png)

Which I rendered as:
[![How to Deploy](https://user-images.githubusercontent.com/4806877/75701933-9c89c580-5c82-11ea-9b1c-1bf267e94d0e.png)](https://docs.google.com/drawings/d/1BqPK92m2cF0wele2S7FXrWeXMvHESSXjc9KuzZC0QUI/edit?usp=sharing)

**Comment by TomAugspurger** [comment]
Thanks for putting these together.

On the first one, I'm not sure  that Dask-ML belongsin the "Data are" blob. dask.delayed *might* fit in there, as it's often used to get data that aren't quite ndarray / dataframe-like into shape.

**Comment by jsignell** [comment]
Yeah, we didn't know where to put dask-ml. It doesn't really need an explanation because if you are doing ML you know it. At the same time though we want to make sure that people know dask-ml exists, so it would be nice to include in this chart.

**Comment by jsignell** [comment]
I'm not sure if these images are interesting or not. But ping @jrbourbeau in case you have any opinions.
